vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/StaffSearchManager.java

```diff
@@ -0,0 +1,221 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import edu.asu.diging.vspace.core.data.TextContentBlockRepository;
+import edu.asu.diging.vspace.core.model.IVSpaceElement;
+import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.ISlideManager;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+import edu.asu.diging.vspace.core.services.IStaffSearchManager;
+
+@Service
+public class StaffSearchManager implements IStaffSearchManager {
+
+    @Value("${page_size}")
+    private int pageSize;
+
+    @Autowired
+    private ISpaceManager spaceManager;
+
+    @Autowired
+    private IModuleManager moduleManager;
+
+    @Autowired
+    private ISlideManager slideManager;
+
+    @Autowired
+    private TextContentBlockRepository textContentBlockRepo;
+    
+    /**
+     * Method to return the requested spaces whose name or description contains the search string
+     * 
+     * @param pageNo. if pageNo<1, 1st page is returned, if pageNo>total pages,last
+     *                page is returned
+     * @return list of spaces whose name or description contains the search string in the requested pageNo and requested order.
+     */
+
+    @Override
+    public HashSet<IVSpaceElement> searchSpaces(String search, int page, long totalSpacePage, String tab,
+            StringBuilder strSpacePageNo) {
+        int spacePageNo;
+        try {
+            spacePageNo = validatePageNumber(page, totalSpacePage);
+        } catch (NumberFormatException numberFormatException) {
+            spacePageNo = 1;
+        }
+        // if it is not the current tab then reset the page number to 1.
+        if (!tab.trim().equals("space")) {
```

Comment:
tab is a front end component and should not be in your business logic (what if we change it to have different pages at some point for instance?)

Reply:
and hard-coding strings is a big no no